code/game/atoms.dm

```diff
@@ -1157,34 +1157,75 @@
   * Must return  parent proc ..() in the end if overridden
   */
 /atom/proc/tool_act(mob/living/user, obj/item/I, tool_type)
+	var/list/processing_recipes = list() //List of recipes that can be mutated by sending the signal
+	. |= SEND_SIGNAL(src, COMSIG_TOOL(tool_type), user, I, processing_recipes)
+	message_admins(COMSIG_TOOL(tool_type))
+	if(processing_recipes.len)
+		process_recipes(user, I, processing_recipes)
 	switch(tool_type)
 		if(TOOL_CROWBAR)
-			. |= crowbar_act(user, I)
+			crowbar_act(user, I)
 		if(TOOL_MULTITOOL)
-			. |= multitool_act(user, I)
+			multitool_act(user, I)
 		if(TOOL_SCREWDRIVER)
-			. |= screwdriver_act(user, I)
+			screwdriver_act(user, I)
 		if(TOOL_WRENCH)
-			. |= wrench_act(user, I)
+			wrench_act(user, I)
 		if(TOOL_WIRECUTTER)
-			. |= wirecutter_act(user, I)
+			wirecutter_act(user, I)
 		if(TOOL_WELDER)
-			. |= welder_act(user, I)
+			welder_act(user, I)
 		if(TOOL_ANALYZER)
-			. |= analyzer_act(user, I)
+			analyzer_act(user, I)
 	if(. & COMPONENT_BLOCK_TOOL_ATTACK)
 		return TRUE
 
-//! Tool-specific behavior procs. They send signals, so try to call ..()
+
+/atom/proc/process_recipes(mob/living/user, obj/item/I, list/processing_recipes)
+	//Only one recipe? use the first
+	if(processing_recipes.len == 1)
+		StartProcessingAtom(user, I, processing_recipes[1])
+		return
+	//Otherwise, select one with a radial
+	ShowProcessingGui(user, I, processing_recipes)
+
+///Creates the radial and processes the selected option
+/atom/proc/ShowProcessingGui(mob/living/user, obj/item/I, list/possible_options)
+	var/list/choicestooptions = list() //Dict of object name | dict of object processing settings
+	var/list/choices = list()
+
+	for(var/i in possible_options)
+		var/list/current_option = i
+		var/atom/current_option_type = current_option[TOOL_PROCESSING_RESULT]
+		choicestooptions[initial(current_option_type.name)] = current_option
+		var/image/option_image = image(icon = initial(current_option_type.icon), icon_state = initial(current_option_type.icon_state))
+		choices += list("[initial(current_option_type.name)]" = option_image)
+
+	var/pick = show_radial_menu(user, src, choices, radius = 36, require_near = TRUE)
+
+	StartProcessingAtom(user, choicestooptions[pick])
+
+
+/atom/proc/StartProcessingAtom(mob/living/user, obj/item/I, list/chosen_option)
+	to_chat(user, "<span class='notice'>You start working on [src]</span>")
```

Comment:
I feel like it might be nice to have an override of sorts to change the vocabulary used here (ie you start slicing the bread) but thats minor